common/src/main/java/io/pravega/common/io/FileModificationWatcher.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.io;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.function.Consumer;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class can be used to create objects that monitor modifications to the specified file and perform a
+ * specified action (in the form of a callback) upon modification detection.
+ *
+ * Note that:
+ * - The specified actions may trigger slightly later than when the file is actually modified.
+ * - If there are multiple file modifications in quick succession, only the last one may trigger the specified action.
+ *
+ */
+@Slf4j
+public class FileModificationWatcher extends Thread {
+
+    /**
+     * The path of file to watch.
+     */
+    private final Path pathOfFileToWatch;
+
+    /**
+     * The action to perform when the specified file changes.
+     */
+    private final Consumer<WatchEvent<?>> callback;
+
+    private final UncaughtExceptionHandler uncaughtExceptionalHandler = (t, e) -> handleException(t.getName(), e);
+
+    public FileModificationWatcher(@NonNull String fileToWatch, @NonNull Consumer<WatchEvent<?>> callback) {
+        super();
+
+        this.pathOfFileToWatch = Paths.get(fileToWatch);
+
+        if (!this.pathOfFileToWatch.toFile().exists()) {
```

Comment:
FileNotFoundException

Reply:
Sure, thanks for noticing it. I've changed it to FileNotFoundException in the latest commit.